processing/src/main/java/io/druid/query/Query.java

```diff
@@ -70,8 +70,18 @@
 
   String getType();
 
+  /**
+   * @deprecated use {@link QueryPlus#run(QuerySegmentWalker, Map)} instead. This method could be removed in the next
```

Comment:
`QueryPlus.run()` is implemented without using this method for `BaseQuery`, i. e. for all query types in the Druid repository (all `Query` impls in the Druid repo extend `BaseQuery`). `QueryPlus.run()` falls back to `Query.run()` only for "unknown" `Query` implementations in extensions. They should be forced to move to the new API when `QueryRunner.run(Query, Map)` method is removed in the next major or minor version of Druid.